### PR TITLE
feat(snapshot): Rename sample -> snapshot, add new sample

### PR DESCRIPTION
**Pull request by briancavalier**
Alternative to #142 

Uses another set of names we've discussed: snapshot, sample, zipItems, withItems.

### Todo

- [x] Update API docs

Close #142

### CODE REVIEW

File: packages/core/src/combinator/zipItems.js [hunk 1]
```diff
@@ -1,31 +1,31 @@
-/** @license MIT License (c) copyright 2010-2017 original author or authors */
+/** @license MIT License (c) copyright 2017 original author or authors */
 
 import { empty } from '../source/empty'
 import Pipe from '../sink/Pipe'
 
-export const withArrayValues = (array, stream) =>
-  zipArrayValues(keepLeft, array, stream)
+export const withItems = (array, stream) =>
```
Comment (Frikki) [inline packages/core/src/combinator/zipItems.js]:
```javascript
export const withItems = (items, stream) =>
  zipItems(keepLeft, items, stream)
```

File: packages/core/src/combinator/zipItems.js [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+  zipItems(keepLeft, array, stream)
 
-export const zipArrayValues = (f, array, stream) =>
+export const zipItems = (f, array, stream) =>
```
Comment (Frikki) [inline packages/core/src/combinator/zipItems.js]:
```javascript
export const zipItems = (f, items, stream) =>
  items.length === 0 || stream === empty()
    ? empty()
    : new ZipItems(f, items, stream)
```

File: packages/core/src/combinator/zipItems.js [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
   array.length === 0 || stream === empty()
     ? empty()
-    : new ZipArrayValues(f, array, stream)
+    : new ZipItems(f, array, stream)
 
 const keepLeft = (a, _) => a
 
-class ZipArrayValues {
+class ZipItems {
   constructor (f, values, source) {
```
Comment (Frikki) [inline packages/core/src/combinator/zipItems.js]:
```javascript
  constructor (f, items, sources) {
    this.f = f
    this.items = items
    this.source = source
  }
```

File: packages/core/src/combinator/zipItems.js [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
     this.f = f
     this.values = values
     this.source = source
   }
 
   run (sink, scheduler) {
-    return this.source.run(new ZipArrayValuesSink(this.f, this.values, sink), scheduler)
+    return this.source.run(new ZipItemsSink(this.f, this.values, sink), scheduler)
```
Comment (Frikki) [inline packages/core/src/combinator/zipItems.js]:
```javascript
  return this.source.run(new ZipItemsSink(this.f, this.items, sink), scheduler)
```

File: packages/core/src/combinator/zipItems.js [hunk 5]
```diff
@@ -1,31 +1,31 @@
...
   }
 }
 
-class ZipArrayValuesSink extends Pipe {
+class ZipItemsSink extends Pipe {
   constructor (f, values, sink) {
```
Comment (Frikki) [inline packages/core/src/combinator/zipItems.js]:
```javascript
  constructor (f, items, sink) {
    super(sink)
    this.f = f
    this.items = items
    this.index = 0
  }

  event (t, b) {
    const f = this.f
    this.sink.event(t, f(this.items[this.index], b))
    
    this.index += 1
    if (this.index >= this.items.length) {
      this.sink.end(t)
    }
  }
```

File: packages/core/test/combinator/zipItems-test.js [hunk 6]
```diff
@@ -2,27 +2,27 @@
...
+import { zipItems, withItems } from '../../src/combinator/zipItems'
 import { empty } from '../../src/source/empty'
 
 import { collectEventsFor, makeEvents } from '../helper/testEnv'
 
-describe('withArrayValues', () => {
-  describe('zipArrayValues', () => {
+describe('zipItems', () => {
+  describe('zipItems', () => {
     it('should be empty for empty array', () => {
```
Comment (Frikki) [inline packages/core/test/combinator/zipItems-test.js]:
```javascript
it('should be empty given an empty item list', () => {
```

File: packages/core/test/combinator/zipItems-test.js [hunk 7]
```diff
@@ -2,27 +2,27 @@
...
       return collectEventsFor(1, s).then(eq([]))
     })
 
     it('should be empty for empty stream', () => {
-      const s = zipArrayValues(fail, [1, 2, 3], empty())
+      const s = zipItems(fail, [1, 2, 3], empty())
       return collectEventsFor(1, s).then(eq([]))
     })
 
     it('should contain zipped values when more events than values', () => {
```
Comment (Frikki) [inline packages/core/test/combinator/zipItems-test.js]:
```javascript
it('should contain zipped items when more events than items', () => {
```

File: packages/core/test/combinator/zipItems-test.js [hunk 8]
```diff
@@ -35,7 +35,7 @@ describe('withArrayValues', () => {
     it('should contain zipped values when more values than events', () => {
```
Comment (Frikki) [inline packages/core/test/combinator/zipItems-test.js]:
```javascript
it('should contain zipped items when more items than events', () => {
```

File: packages/core/test/combinator/zipItems-test.js [hunk 9]
```diff
@@ -45,21 +45,21 @@ describe('withArrayValues', () => {
     })
   })
 
-  describe('withArrayValues', () => {
+  describe('withItems', () => {
     it('should be empty for empty array', () => {
```
Comment (Frikki) [inline packages/core/test/combinator/zipItems-test.js]:
```javascript
it('should be empty given an empty item list', () => {
```

File: packages/core/test/combinator/zipItems-test.js [hunk 10]
```diff
@@ -45,21 +45,21 @@ describe('withArrayValues', () => {
...
       return collectEventsFor(1, s).then(eq([]))
     })
 
     it('should be empty for empty stream', () => {
-      const s = withArrayValues([1, 2, 3], empty())
+      const s = withItems([1, 2, 3], empty())
       return collectEventsFor(1, s).then(eq([]))
     })
 
     it('should contain array values when more events than values', () => {
```
Comment (Frikki) [inline packages/core/test/combinator/zipItems-test.js]:
```javascript
it('should contain items when more events than items', () => {
```

File: packages/core/test/combinator/zipItems-test.js [hunk 11]
```diff
@@ -72,7 +72,7 @@ describe('withArrayValues', () => {
     it('should contain array values when more values than events', () => {
```
Comment (Frikki) [inline packages/core/test/combinator/zipItems-test.js]:
```javascript
it('should contain items when more items than events', () => {
```

File: packages/core/src/combinator/zipItems.js (hunk 1 above)
Comment (briancavalier) [reply packages/core/src/combinator/zipItems.js]: :+1: Excellent.  I'll tackle all these "values" -> "items" changes at once.

**Comment by briancavalier** [comment]
🧀

**Comment by Frikki** [comment]
_Click! Click! Click! Click! Click!_

**Comment by Frikki** [comment]
Be aware of PR #149, which introduce changes to the same code. :cold_sweat:

**Pull request merged**